src/settings.cc

```diff
@@ -202,6 +202,16 @@ void Settings::visit(SettingsVisitor& visitor)
 	}
 }
 
+SettingsEntry* Settings::getSettingEntryByName(std::string name)
+{
+	for (std::list<SettingsEntry *>::iterator it = entries.begin();it != entries.end();it++) {
```

Comment:
Since C++11, you can also do something like `for (SettingsEntry* entry : entries) {` and avoid having to deal with iterators directly.

Reply:
Agreed - prefer C++11 for all new code.

Reply:
Going down the C++11 route, this would be even better: `for (auto entry : entries) {`

Reply:
I try to match the preexisting code style, in this case the visit function.
Well: changed for my function, preparing a pull-request for the visit function in the master branch.
(changing the visit function in a feature branch would not make much sense)

Reply:
Yup, sometimes it's easier to follow existing style, sometimes it's easier to update all of it once touching a file. We made the switch to C++11 a while back, but there is still some pieces of code lagging behind.